library/azure_rm_resource.py

```diff
@@ -0,0 +1,296 @@
+      body: "{{ body }}"
+'''
+
+RETURN = '''
+response:
+    description: Response specific to resource type.
+    returned: always
+    type: dict
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure import AzureConfiguration
+    from msrest.service_client import ServiceClient
+    import json
+
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMResource(AzureRMModuleBase):
+    def __init__(self):
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            url=dict(
+                type='str',
+                required=False
+            ),
+            provider=dict(
+                type='str',
+                required=False
+            ),
+            resource_group=dict(
+                type='str',
+                required=False
+            ),
+            resource_type=dict(
+                type='str',
+                required=False
+            ),
+            resource_name=dict(
+                type='str',
+                required=False
+            ),
+            subresource_type=dict(
+                type='str',
+                required=False
+            ),
+            subresource_name=dict(
+                type='str',
+                required=False
+            ),
+            api_version=dict(
+                type='str',
+                required=True
+            ),
+            method=dict(
+                type='str',
+                default='PUT',
+                choices=[ "GET", "PUT", "POST", "HEAD", "PATCH", "DELETE", "MERGE" ]
+            ),
+            body=dict(
+                type='raw'
+            ),
+            status_code=dict(
+                type='list',
+                default=[200, 202]
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+        # store the results of the module operation
+        self.results = dict(
+            changed=False,
+            response=None
+        )
+        self.mgmt_client = None
+        self.url = None
+        self.api_version = None
+        self.provider = None
+        self.resource_group = None
+        self.resource_type = None
+        self.resource_name = None
+        self.subresource_type = None
+        self.subresource_name = None
+        self.method = None
+        self.status_code = []
+        self.state = None
+        super(AzureRMResource, self).__init__(self.module_arg_spec)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(GenericRestClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        if self.state == 'absent':
```

Comment:
didn't get the point, when state='absent', then delete

Reply:
that's how Ansible modules work:
- present (default) - resource should be created or updated
- absent - resource should be absent after module is run, meaning deleted.
my intention here is to make module as similar as possible to Ansible pattern (despite it's generic module).
So, by default, method will be PUT if present and DELETE if absent.
User may still use method parameter to override PUT with POST or anything else in some cases.